pkg/cli/alpha.go

```diff
@@ -21,7 +21,7 @@ import (
 	"strings"
 
 	"github.com/spf13/cobra"
-	configgen "sigs.k8s.io/kubebuilder/v3/pkg/cli/alpha/config-gen"
+	configgen "sigs.k8s.io/kubebuilder/v3/pkg/plugins/config-gen"
 )
```

Comment:
the idea of https://github.com/kubernetes-sigs/kubebuilder/issues/2501, would be we change it to follow up the [Plugin architecture](https://book.kubebuilder.io/plugins/plugins.html). 

In this way, its acceptance criteria/what we need to do would be like; 

- scaffold the project following the config-gen layout when we run `kubebuilder init --plugins=config-gen/alpha1`
- required to figure out what is the best behaviour, but check what should be done if we run `kubebuilder create API when the scaffold done was using config-gen
- config-gen alpha command migrated to a alpha plugin
- Its sample scaffolded under testdata (see that we do the commands calls to build the sample in https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh)
- It can be a follow-up - but we also need to figure out its docs. We might need a new section under https://book.kubebuilder.io/plugins/plugins.html to begin to document all plugins which are implemented in this repo.

Reply:
Hey, @camilamacedo86 ,
Glad to have your response. Thx for providing me with contexts in details!
If I'm understanding correct, we are expecting to migrate `kb alpha config-gen ...` to `kb init --plugins=config-gen/alpha1`. That seems like a breaking change where it is likely to bring users more convenience to config their controller manager:
I assume when executing `kb create api`, a new folder of an empty `kubebuilderconfiggen.yaml` can be created.
Users can configure their controllers over there.
And so they don't have to manually apply the 'output yaml file'. Instead, `make deploy` should load `kubebuilderconfiggen.yaml` automatically and deploy the controller corresponding to users' configuration.
Is this what we are expecting for?

Reply:
Hi, camila, some follow ups:
1. Considering about the migration destination, would you think `pkg/plugins/config-gen/alpha` be a good place for the code base in respect to an `alpha plugin`? (This seems similar to how current plugins directories are structured..)
2. For the layout generated by `kubebuilder init --plugins=config-gen/alpha1`, do we expect a folder containing `kubebuildconfiggen.yaml` under `config/`? I'm not sure if I'm understanding correct.. Well, the case is we are currently having `config/manager` that composes the yaml file for deploying controller. Maybe it is confusing to have the config-gen configs in parallel with `config/manager` in the same folder?
3. How do we expect to use this plugin once the cmd migrated? On the previous thread above, I was thinking if we want to integrate it into `make deploy`. And...when I pick a view on [the origin talking about `configgen`](https://www.youtube.com/watch?v=mb-aAsZ0-ws&list=PL69nYSiGNLP0SOaFeA9f3dwdCNECEKkX3&index=33&ab_channel=SollyRoss), I realized that it seems an early stage for `config-gen`, maybe it is not fully prepared to integrate it into the main workflow? So I am wondering if it is enough to provide a command to generate yaml file and let users to decide how to use that. If so, is it good to add additional scripts in `Makefile` to generate customized yaml when `init` with the plugin?

Reply:
Hi @Kavinjsir, 

For 1.: For me, that shows fine. +1

For 2 and 3.

Currently, it uses the cmd command. Instead of that, we will need to ensure that when this plugin is used in the init we do not scaffold the kustomize one by default and we scaffold all that is required for the project work. 

See that we have a PR that was not finished but init this work: https://github.com/kubernetes-sigs/kubebuilder/pull/2099. You might continue this work.